extensions/cloudcredentials/cloudcredentials.go

```diff
@@ -23,3 +31,54 @@ type CloudCredential struct {
 	VmwareVsphereConfig          *VmwarevsphereCredentialConfig `json:"vmwarevspherecredentialConfig,omitempty"`
 	UUID                         string                         `json:"uuid,omitempty"`
 }
+
+func LoadCloudCredential(provider string) CloudCredential {
+	var cloudCredential CloudCredential
+	switch {
+
+	case provider == provisioninginput.AWSProviderName.String():
+		var awsCredentialConfig AmazonEC2CredentialConfig
+		config.LoadConfig(AmazonEC2CredentialConfigurationFileKey, &awsCredentialConfig)
+		cloudCredential.AmazonEC2CredentialConfig = &awsCredentialConfig
+		return cloudCredential
```

Comment:
nit - can you add a new line  after 42? So that the return is not so consolidated? Same for the rest in this switch statement.